### PR TITLE
bug: Serialize Decimal correctly for ios aps

### LINKED ISSUE #899: Serialize Decimal values in iOS APNs output
See https://sentry.prod.mozaws.net/operations/autopush-dev/issues/605747/

### PR DESCRIPTION
**Pull request by jrconlin**
closes #899

### CODE REVIEW

File: autopush/router/apns2.py [hunk 1]
```diff
@@ -21,6 +22,16 @@
 APNS_PRIORITY_LOW = '5'
 
 
+class ComplexEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, Decimal):
+            return int(float(obj.to_integral()))
```
Comment (bbangert) [inline autopush/router/apns2.py]: Why not use to_integral_value? The docs indicate it should already return an int.

Comment (jrconlin) [inline autopush/router/apns2.py]: I'll blame pycharm a bit for this one. When I just used object, i was getting warnings. To be safe, i figured being explicit would be safer.

Comment (jrconlin) [inline autopush/router/apns2.py]: Going to switch this out for less juggling

Comment (jrconlin) [inline autopush/router/apns2.py]: Hrm, ok, so reading the docs and thinking about this, there are a few possible methods that I should take. While the outcome is "dump this out as an integer" and this is absolutely temporary while the iOS folk are poking at it, there's a few things at play. 

`to_integral` (or properly `to_integral_value`) does the rounding inside the decimal without flagging an exception. float converts the decimal to a numeric and int converts to the expected JSON serializable element. Having decimal do it's rounding is probably the most accurate thing. The other two steps are pretty much required.

Comment (jrconlin) [inline autopush/router/apns2.py]: And, this is why reading a code review comment while following a keynote is probably not a good idea. Misread your comment. *sigh*

File: autopush/router/apns2.py [hunk 2]
```diff
@@ -21,6 +22,16 @@
 APNS_PRIORITY_LOW = '5'
 
 
+class ComplexEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, Decimal):
+            return int(float(obj.to_integral_value()))
```
Comment (pjenvey) [inline autopush/router/apns2.py]: should kill the float() here: to_integral_value already clips it

Comment (jrconlin) [inline autopush/router/apns2.py]: Ah, thanks!

**Pull request merged**